File: String/Defanging_an_ip_address.cpp
class Solution {
public:
    string defangIPaddr(string address) {
        int len = address.length();
        string str;
        /* Appending to String */
        for(int i=0;i<len;i++)
        { 
            if(address[i] == '.')
            {
                str += '[';
                str += '.';
                str += ']';
            }
            else
                str += address[i];
        }
        return str;
    }
};

File: String/Java/Easy/Defanging_an_ip_address.java
class Solution {
    public String defangIPaddr(String address) {
        char[] array = address.toCharArray();
        StringBuilder str = new StringBuilder();
        for(char c: array)
        {
            if(c == '.')
            {
                str.append("[.]");
            }
            else{
                str.append(c);
            }
        }
        return str.toString();
        
    }
}

/* -------------- OR -------------- */

class Solution {
    public String defangIPaddr(String address) {
        String defangedIp = address.replaceAll("\\.","[.]");
        return defangedIp;
        
    }
}

/* -------------- OR -------------- */
class Solution {
    public String defangIPaddr(String address) {
        String defangedIp = address.replace(".","[.]");
        return defangedIp;
        
    }
}

/* -------------- OR -------------- */

class Solution {
    public String defangIPaddr(String address) {
        String defangedIp = String.join("[.]",address.split("\\."));
        return defangedIp;
        
    }
}